cassandra/connection.py

```diff
@@ -309,16 +309,17 @@ def __repr__(self):
 
 class SniEndPointFactory(EndPointFactory):
 
-    def __init__(self, proxy_address, port):
+    def __init__(self, proxy_address, port, node_domain=None):
         self._proxy_address = proxy_address
         self._port = port
+        self._node_domain = node_domain
 
     def create(self, row):
         host_id = row.get("host_id")
         if host_id is None:
             raise ValueError("No host_id to create the SniEndPoint")
-
-        return SniEndPoint(self._proxy_address, str(host_id), self._port)
+        address = "{}.{}".format(host_id, self._node_domain) if self._node_domain else str(host_id)
```

Comment:
node_domain shouldn't be empty, so there shouln't be conditional value here, it's always host_id + node_domain